src/content/protocols/kodiak-v3.md

```diff
@@ -0,0 +1,230 @@
+---
+protocol: "Kodiak Finance V3"
+website: "https://www.kodiak.finance/"
+x: "https://x.com/KodiakFi"
+github: ["https://github.com/Kodiak-Finance"]
+defillama_slug: ["kodiak-v3"](https://defillama.com/protocol/kodiak-v3)
+chain: "Berachain"
+stage: 0
+reasons: []
+risks: ["L", "H", "H", "H", "H"]
+author: ["gnart33"]
+submission_date: "2025-05-07"
+publish_date: "2025-05-07"
+update_date: "2025-05-07"
+---
+
+# Summary
+Kodiak Finance V3 is a comprehensive liquidity hub on Berachain that combines multiple DeFi components into a unified platform. At its core, it offers both Uniswap V2-style (constant product) and V3-style (concentrated liquidity) AMMs, with a protocol fee of 35% from launch. The platform's unique "Kodiak Islands" system provides ERC20-wrapped V3 positions in two variants: Managed Islands, which are protocol-run with automatic rebalancing and a 10% fee, and Unmanaged Islands that allow users to deploy their own positions without rebalancing.
+A distinctive feature of Kodiak is its integrated incentive layer and no-code token deployer factory called "Panda Factory." This system enables token launches on Berachain using bonding curves with several innovative features: a 1B total supply model, zero initial liquidity requirement, configurable price ranges, and support for base tokens like BERA and HONEY. The Panda Factory includes an optional same-block buy feature for deployers (up to 10% of supply) and implements a graduation mechanism where tokens transition from the bonding curve to the Kodiak DEX once 80-90% of the supply is purchased.
+The protocol's architecture is built on immutable contracts rather than using proxy upgradeable contracts, with protocol parameters managed through multi-sig wallets.
+# Overview
+
+## Chain
+
+Berachain Mainnet, an EVM-Identical Layer 1 blockchain utilizing Proof-of-Liquidity 
+
+> Chain score: Low
+
+## Upgradeability
```

Comment:
upgradeability risk is defined as

1)
Possible upgrades may result in the theft or loss of user funds -> high risk

2)
Possible upgrades may result in the theft or loss of unclaimed yield or may otherwise materially change the expected performance of the system (but user funds remain unaffected) -> medium risk

Please go over the permissions table, once this and the system diagram overview are good, the upgradeability section is easier to write

Reply:
please refrain from generic statements like 

> but malicious actions could lead to unexpected behavior of the system.